shapeshift/web
# feat: report spam NFTs

## Description

Adds a "Report Spam" button marking the collection as spam and hiding it in the NFT slice, as well as reporting the NFT as spam to NFT, as long as the NFT is from a supported chain for the `reportSpam` endpoint: https://docs.alchemy.com/reference/reportspam i.e Ethereum, Mainnet, and Polygon.

## Pull Request Type


## Issue (if applicable)



closes https://github.com/shapeshift/web/issues/5075

## Risk



Theoretically none - worst that happens, we fire a malformed request to Alchemy or don't fire a request to report the spammy NFT collection when we should, though tested and this does what it says on the box.

## Testing



- Click the "Report Spam" button
- Ensure a network request is made to Alchemy, and a log with the response from Alchemy appears in the console
- Ensure the modal closes after the report, and the collection NFT/s is/are now gone

### Engineering



- ☝🏽 

### Operations



- ☝🏽 

## Screenshots (if applicable)

- Develop

<img width="1728" alt="image" src="https://github.com/shapeshift/web/assets/17035424/c4af92b8-ffc5-45a9-b592-2297820faa5d">
<img width="1085" alt="image" src="https://github.com/shapeshift/web/assets/17035424/9dbfc20d-c6b9-4823-8406-95f60bf5de1d">
<img width="1372" alt="image" src="https://github.com/shapeshift/web/assets/17035424/72c491b3-ea59-4849-bb26-fbdd321d1d19">

- NFT reporting

<img width="858" alt="image" src="https://github.com/shapeshift/web/assets/17035424/cf21ae14-7ba5-4bcf-afe7-cc5b648c2a73">


- This diff - all spam NFTs reported

<img width="1728" alt="image" src="https://github.com/shapeshift/web/assets/17035424/cf062499-d7ac-4a4a-bda7-49afafa787a1">
<img width="1408" alt="image" src="https://github.com/shapeshift/web/assets/17035424/9aebdfe2-fa01-4e29-8ec9-8fc6728c98d8">
